daemon/cluster/cluster.go

```diff
@@ -725,6 +730,24 @@ func (c *Cluster) UpdateService(serviceID string, version uint64, spec types.Ser
 		return err
 	}
 
+	if encodedAuth != "" {
+		serviceSpec.Task.Runtime.(*swarmapi.TaskSpec_Container).Container.PullOptions = &swarmapi.ContainerSpec_PullOptions{RegistryAuth: encodedAuth}
+	} else {
+		// this is needed because if the encodedAuth isn't being updated then we
+		// shouldn't lose it, and continue to use the one that was already present
+		currentService, err := getService(c.getRequestContext(), c.client, serviceID)
+		if err != nil {
+			return err
+		}
+		ctnr := currentService.Spec.Task.GetContainer()
+		if ctnr == nil {
+			return fmt.Errorf("Service does not use container tasks")
```

Comment:
Generally error values are supposed to start with a lowercase character in Go. I guess we are not very good about following this convention in Docker, though. So if you want to keep it this way for consistency with the other error messages, that's fine with me.